File: 3324-find-the-sequence-of-strings-appeared-on-the-screen/3324-find-the-sequence-of-strings-appeared-on-the-screen.py
class Solution:
    def stringSequence(self, target: str) -> List[str]:
        current = ''
        result = []
        
        for char in target:
            if not current:
                current += 'a'
                result.append(current)
            
            while current[-1] != char:
                current = current[:-1] + chr((ord(current[-1]) - ord('a') + 1) % 26 + ord('a'))
                result.append(current)
            
            if len(current) < len(target):
                current += 'a'
                result.append(current)
        
        return result

File: 3324-find-the-sequence-of-strings-appeared-on-the-screen/3324-find-the-sequence-of-strings-appeared-on-the-screen.java
class Solution {
    public List<String> stringSequence(String target) {
        ArrayList<String> list = new ArrayList<>();
        for (int i = 0; i < target.length(); i++) {
            for (char c = 'a'; c <= target.charAt(i); c++) {
                list.add(target.substring(0, i) + c);
            }
        }
        return list;
    }
}